tools/RAiDER/models/hrrr.py

```diff
@@ -134,152 +112,82 @@ def load_weather(self, *args, filename=None, **kwargs):
-        lons = lons[y_min:y_max, x_min:x_max]
-
-        # Data variables
-        t = ds['t'][:, y_min:y_max, x_min:x_max].to_numpy()
-        z = ds['gh'][:, y_min:y_max, x_min:x_max].to_numpy()
-        q = ds['q'][:, y_min:y_max, x_min:x_max].to_numpy()
-        ds.close()
-
-        # This section is purely for flipping arrays as needed
-        # to match ECMWF reader is doing
-        # All flips are views - no extra memory use
-        # Lon -> From west to east
-        # Lat -> From south to north (ECMWF reads north to south and flips it
-        # load_weather) - So we do south to north here
-        # Pres -> High to Los - (ECWMF does now to high and flips it back) - so
-        # we do high to low
-        # Data is currently in [levels, y, x] order
-        flip_axes = []
-        if levels[-1] > levels[0]:
-            flip_axes.append(0)
-            levels = np.flip(levels)
-
-        if lats[0, 0] > lats[-1, 0]:
-            flip_axes.append(1)
-            lats = np.flip(lats, 0)
-            yArr = np.flip(yArr)
-
-        if lons[0, 0] > lons[0, -1]:
-            flip_axes.append(2)
-            lons = np.flip(lons, 1)
-            xArr = np.flip(xArr)
-
-        flip_axes = tuple(flip_axes)
-        if flip_axes:
-            t = np.flip(t, flip_axes)
-            z = np.flip(z, flip_axes)
-            q = np.flip(q, flip_axes)
-
-        # Create output dataset
-        ds_new = xarray.Dataset(
-            data_vars=dict(
-                t= (["level", "y", "x"], t,
-                    {"grid_mapping": "proj"}),
-                z= (["level", "y", "x"], z,
-                    {"grid_mapping": "proj"}),
-                q= (["level", "y", "x"], q,
-                    {"grid_mapping": "proj"}),
-                lats=(["y", "x"], lats),
-                lons=(["y", "x"], lons),
-            ),
-            coords=dict(
-                levels=(["level"], levels,
-                       {"units": "millibars",
-                        "long_name":  "pressure_level",
-                        "axis": "Z"}),
-                x=(["x"], xArr,
-                   {"standard_name": "projection_x_coordinate",
-                    "units": "m",
-                    "axis": "X"}),
-                y=(["y"], yArr,
-                   {"standard_name": "projection_y_coordinate",
-                    "units": "m",
-                    "axis": "Y"}),
-            ),
-            attrs={
-                'Conventions': 'CF-1.7',
-                'Weather_model':'HRRR',
-           }
-        )
-
-        # Write projection of output
-        ds_new["proj"] = int()
-        for k, v in self._proj.to_cf().items():
-            ds_new.proj.attrs[k] = v
-
-        ds_new.to_netcdf(out, engine='netcdf4')
-
-
     def _download_hrrr_file(self, DATE, out, model='hrrr', product='prs', fxx=0, verbose=False):
         '''
         Download a HRRR model
         '''
         ## TODO: Check how Herbie does temporal interpolation
```

Comment:
Is this still needed here are you not picking the exact model outputs. Where are we doing temporal interpolation?